libs/agno/agno/agent/agent.py

```diff
@@ -3187,23 +3187,25 @@ def _handle_model_response_chunk(
                         run_response.content = model_response.content
                         run_response.content_type = "str"
 
-                if model_response_event.thinking is not None:
-                    model_response.thinking = (model_response.thinking or "") + model_response_event.thinking
-                    run_response.thinking = model_response.thinking
-
                 if model_response_event.reasoning_content is not None:
                     model_response.reasoning_content = (
                         model_response.reasoning_content or ""
                     ) + model_response_event.reasoning_content
                     run_response.reasoning_content = model_response.reasoning_content
 
-                if model_response_event.redacted_thinking is not None:
-                    model_response.redacted_thinking = (
-                        model_response.redacted_thinking or ""
-                    ) + model_response_event.redacted_thinking
+                if model_response_event.reasoning_content is not None:
+                    if not model_response.reasoning_content:
+                        model_response.reasoning_content = model_response_event.reasoning_content
+                    else:
+                        model_response.reasoning_content += model_response_event.reasoning_content
+                    run_response.reasoning_content = model_response.reasoning_content
 
-                    # We only have thinking on response
-                    run_response.thinking = model_response.redacted_thinking
+                if model_response_event.redacted_reasoning_content is not None:
+                    if not model_response.reasoning_content:
+                        model_response.reasoning_content = model_response_event.redacted_reasoning_content
+                    else:
+                        model_response.reasoning_content += model_response_event.redacted_reasoning_content
```

Comment:
Should this not add to the redacted reasoning content?

Reply:
We decided to keep it in `reasoning_content` to avoid carrying provider-specific fields. Later we may add a `provider_specific` kind of field to carry all these.